x-pack/plugins/ml/public/application/model_management/add_model_flyout.tsx

```diff
@@ -0,0 +1,332 @@
+                        />
+                      </p>
+                    </EuiText>
+                  ),
+                },
+                {
+                  title: i18n.translate(
+                    'xpack.ml.trainedModels.addModelFlyout.thirdParty.step4Title',
+                    {
+                      defaultMessage: 'Deploy your model',
+                    }
+                  ),
+                  children: (
+                    <EuiText>
+                      <p>
+                        <FormattedMessage
+                          id="xpack.ml.trainedModels.addModelFlyout.thirdParty.step4Body"
+                          defaultMessage="Press the play button or click “Start deployment” in the drop menu in the table row containing your new model to deploy and use it."
+                        />
+                      </p>
+                    </EuiText>
+                  ),
+                },
+              ]}
+            />
+          </>
+        ),
+      },
+    ];
+  }, [
+    docLinks.links.clients.eland,
+    docLinks.links.ml.nlpImportModel,
+    modelDownloads,
+    selectedModelId,
+  ]);
+
+  const selectedTabContent = useMemo(() => {
+    return tabs.find((obj) => obj.id === selectedTabId)?.content;
+  }, [selectedTabId, tabs]);
+
+  return (
+    <EuiFlyout ownFocus onClose={onClose} aria-labelledby={'addTrainedModelFlyout'}>
+      <EuiFlyoutHeader hasBorder>
+        <EuiTitle size="m">
+          <h2 id={'addTrainedModelFlyout'}>
+            <FormattedMessage
+              id="xpack.ml.trainedModels.addModelFlyout.title"
+              defaultMessage="Add a trained model"
+            />
+          </h2>
+        </EuiTitle>
+      </EuiFlyoutHeader>
+      <EuiFlyoutBody>
+        <EuiTabs>
+          {tabs.map((tab) => (
+            <EuiTab
+              isSelected={selectedTabId === tab.id}
+              onClick={setSelectedTabId.bind(null, tab.id)}
+            >
+              {tab.name}
+            </EuiTab>
+          ))}
+        </EuiTabs>
+        <EuiSpacer size={'m'} />
+        {selectedTabContent}
+      </EuiFlyoutBody>
+      <EuiFlyoutFooter>
+        <EuiFlexGroup justifyContent="spaceBetween">
+          <EuiFlexItem grow={false}>
+            <EuiButtonEmpty iconType="cross" onClick={onClose} flush="left">
+              <FormattedMessage
+                id="xpack.ml.trainedModels.addModelFlyout.closeButtonLabel"
+                defaultMessage="Close"
+              />
+            </EuiButtonEmpty>
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiButton
+              onClick={onSubmit.bind(null, selectedModelId!)}
+              fill
+              disabled={!selectedModelId}
+            >
+              <FormattedMessage
+                id="xpack.ml.trainedModels.addModelFlyout.downloadButtonLabel"
+                defaultMessage="Download"
```

Comment:
@mdefazio I'm wondering about the placement of the Download button in the footer. If you switch to the Third-party tab (with the Eland instructions), the Download button remains visible and enabled according to the selection in the first ELSER tab, so it might not be obvious what is going to be downloaded.  Or worse still, as it's a primary button you could mistakenly click on the Download button and unintentionally start a download you didn't want.

Should this button be disabled or hidden on switching to the Third-party tab?

<img width="857" alt="image" src="https://github.com/elastic/kibana/assets/7405507/bbf1d8bb-e25a-4277-a1ce-15d4501b4e57">

Reply:
The layout has been updated in the latest changes. The download button resides in underneath the models list.